src/main/java/org/opensearch/security/auth/internal/NoOpAuthenticationBackend.java

```diff
@@ -46,7 +46,9 @@ public String getType() {
 
     @Override
     public User authenticate(final AuthCredentials credentials) {
-        return new User(credentials.getUsername(), credentials.getBackendRoles(), credentials);
+        User user = new User(credentials.getUsername(), credentials.getBackendRoles(), credentials);
+        user.addSecurityRoles(credentials.getSecurityRoles());
```

Comment:
Why was this needed?

Reply:
AuthCredential class didn't have mechanism of passing the security roles, and it only had backend roles included before. So this is the way that the obo token can have security roles included in its payload.